android-n/CommitContentSampleApp/CommitContentSampleApp/MainActivity.cs

```diff
@@ -0,0 +1,292 @@
+					inputContentInfo.RequestPermission();
+				}
+				catch (Java.Lang.Exception e)
+				{
+					Log.Error(Tag, "InputContentInfoCompat#requestPermission() failed.", e);
+					return false;
+				}
+			}
+
+			mMimeTypes.Text = string.Join(".", inputContentInfo.Description.FilterMimeTypes("*/*"));
+			mContentUri.Text = inputContentInfo.ContentUri.ToString();
+			mLabel.Text = inputContentInfo.Description.Label;
+			Android.Net.Uri linkUri = inputContentInfo.LinkUri;
+			mLinkUri.Text = linkUri != null ? linkUri.ToString() : "null";
+			mFlags.Text = FlagsToString(flags);
+			mWebView.LoadUrl(inputContentInfo.ContentUri.ToString());
+			mWebView.SetBackgroundColor(Color.Transparent);
+
+			// Due to the asynchronous nature of WebView, it is a bit too early to call
+			// inputContentInfo.releasePermission() here. Hence we call IC#releasePermission() when this
+			// method is called next time.  Note that calling IC#releasePermission() is just to be a
+			// good citizen. Even if we failed to call that method, the system would eventually revoke
+			// the permission sometime after inputContentInfo object gets garbage-collected.
+			mCurrentInputContentInfo = inputContentInfo;
+			mCurrentFlags = flags;
+
+			return true;
+		}
+
+		protected override void OnSaveInstanceState(Bundle outState)
+		{
+			if (mCurrentInputContentInfo != null)
+			{
+				outState.PutParcelable(InputContentInfoKey, (IParcelable) mCurrentInputContentInfo.Unwrap());
+				outState.PutInt(CommitContentFlagsKey, mCurrentFlags);
+			}
+			mCurrentInputContentInfo = null;
+			mCurrentFlags = 0;
+			base.OnSaveInstanceState(outState);
+		}
+
+		/**
+		 * Creates a new instance of {@link EditText} that is configured to specify the given content
+		 * MIME types to EditorInfo#contentMimeTypes so that developers can locally test how the current
+		 * input method behaves for such content MIME types.
+		 *
+		 * @param contentMimeTypes A {@link String} array that indicates the supported content MIME
+		 *                         types
+		 * @return a new instance of {@link EditText}, which specifies EditorInfo#contentMimeTypes with
+		 * the given content MIME types
+		 */
+		private EditText CreateEditTextWithContentMimeTypes(string[] contentMimeTypes)
+		{
+			string hintText;
+			string[] mimeTypes;  // our own copy of contentMimeTypes.
+			if (contentMimeTypes == null || contentMimeTypes.Length == 0)
+			{
+				hintText = "MIME: []";
+				mimeTypes = new string[0];
+			}
+			else
+			{
+				hintText = "MIME: " + string.Join(".", contentMimeTypes);
+				mimeTypes = new string[contentMimeTypes.Length];
+				contentMimeTypes.CopyTo(mimeTypes, 0);
+			}
+
+			var editText = new CustomEditText(this) {MimeTypes = mimeTypes, Hint = hintText};
+			editText.SetTextColor(Color.White);
+			editText.SetHintTextColor(Color.White);
+			return editText;
+		}
+
+		private class CustomEditText : EditText
+		{
+			public string[] MimeTypes { get; set; }
+			public MainActivity Owner { get; set; }
```

Comment:
'Owner' is never set and will throw when we try to invoke OnCommitContentListenerImpl.OnCommitContent. One solution is below:

```
public CustomEditText(MainActivity context) : base(context)
{
	Owner = context;
}
```

Also the three additional constructor declarations are not needed.